portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -55,19 +58,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     
     // Create the query and prepared query to load comment entities from database
     Query query = new Query(COMMENT).addSort(TIMESTAMP_PROPERTY, SortDirection.DESCENDING);
-
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
-    
-    // Add limited number comment contents to the msglist  
-    ArrayList<String> msglist = new ArrayList<String>();
+
+    // Add limited number of comment contents and user email to the userComment Map  
+    HashMap<String, ArrayList<String>> userComment = new HashMap<String, ArrayList<String>>();
     for (Entity commentEntity:results.asList(FetchOptions.Builder.withLimit(maxNumComments))) {
-      String commentMsg = (String)commentEntity.getProperty(CONTENT_PROPERTY);
-      msglist.add(commentMsg);
+        String commentMsg = (String)commentEntity.getProperty(CONTENT_PROPERTY);
+        String commentUser= (String)commentEntity.getProperty(USEREMAIL_PROPERTY);
+        ArrayList<String> msglist = userComment.get(commentUser);
+        if (msglist == null) {
+            msglist = new ArrayList<String>();
+            msglist.add(commentMsg);
+            userComment.put(commentUser, msglist);
```

Comment:
Suggestion: We can reduce the # of lines if you do:
userComment.put(commentUser, new ArrayList<String>(commentMsg));

Reply:
new ArrayList(commentMsg) is not working for Java 8. But I changed it by initializing in a line like following: new ArrayList<String>(Arrays.asList(commentMsg))